tests/msc2716_test.go

```diff
@@ -610,12 +620,11 @@ func TestBackfillingHistory(t *testing.T) {
 				200,
 			)
 			batchSendResBody := client.ParseJSON(t, batchSendRes)
-			historicalEventIDs := getEventsFromBatchSendResponseBody(t, batchSendResBody)
-			baseInsertionEventID := historicalEventIDs[len(historicalEventIDs)-1]
+			historicalEventIDs := getEventIDsFromBatchSendResponseBody(t, batchSendResBody)
+			baseInsertionEventID := getBaseInsertionEventIDFromBatchSendResponseBody(t, batchSendResBody)
 
-			// [1 insertion event + 2 historical events + 1 chunk event + 1 insertion event]
-			if len(historicalEventIDs) != 5 {
-				t.Fatalf("Expected eventID list should be length 5 but saw %d: %s", len(historicalEventIDs), historicalEventIDs)
+			if len(historicalEventIDs) != 2 {
+				t.Fatalf("Expected eventID list should be length 2 but saw %d: %s", len(historicalEventIDs), historicalEventIDs)
```

Comment:
Please please please document why you expect this to be 2. I don't see you changing the number of injected events, so why shouldn't this be 5? I'm guessing the event IDs returned in the batch endpoint are the earliest or something? But this is really really not clear. It's hard enough to follow all the event IDs and markers and timestamps without randomly changing a 5 into a 2 without any reason why.

Reply:
What's changed here is we have gone from including the 3 insertion and chunk events in the `event_ids` response, to separating them out to their own fields.

So now `event_ids` in the response only includes the 2 historical events that you sent in your request which is much more obvious for a downstream user (spawned from https://github.com/matrix-org/matrix-doc/pull/2716#discussion_r684765034)

I've updated the assertion message to make it clear `Expected 2 event_ids in the response that correspond to the 2 events we sent in the request but saw XX: ...`